src/components/SingleList.jsx

```diff
@@ -1,13 +1,41 @@
 import './SingleList.css';
+import { useState } from 'react';
+import PushPinIcon from '@mui/icons-material/PushPin';
 
-export function SingleList({ name, path, setListPath }) {
-	function handleClick() {
+export function SingleList({
+	name,
+	path,
+	setListPath,
+	setImportantList,
+	isImportant,
+}) {
+	const [isHovered, setIsHovered] = useState(false);
+
+	const handleClick = () => {
 		setListPath(path);
-	}
+	};
+
+	const handleImportantList = () => {
+		setImportantList(path);
+	};
+
+	const pinStyle = isImportant ? 'visible' : !isHovered ? 'hidden' : 'visible';
 
```

Comment:
pin is displayed based on whether it is: 
- important, 
- not hovered,
- hovered